amazon-kinesis-client/src/test/java/software/amazon/kinesis/utils/RecordValidatorQueue.java

```diff
@@ -0,0 +1,66 @@
+package software.amazon.kinesis.utils;
+
+import lombok.extern.slf4j.Slf4j;
+import java.util.*;
+
+@Slf4j
+public class RecordValidatorQueue {
+
+    HashMap<String, List<String>> dict = new HashMap<>();
```

Comment:
3. The way that this class is being used is that each thread "puts" records to a different key (shardId) until all the threads are completed. Only after all threads have finished is anything read from the HashMap. Because all the writes are to different keys and no reads are done until all the threads are done, I did not think that any race conditions were possible here. There may be a case that I am missing, however.